src/js/modules/supervisors/FileManager.ts

```diff
@@ -145,7 +149,7 @@ class FileManager {
    * @param repository, is a coprocessor container
    */
   updateRepositoryOnNewFile(repository: Repository): void {
-    return this.submitDirWatcher.on("add", (filePath) => {
+    this.submitDirWatcher.on("add", (filePath) => {
```

Comment:
Shouldn't the watch be set in the constructor? I think during construction it makes sense to define once what files and what events for those files you are watching

Reply:
Putting the set on a public method means to me that someone can call .on() more than once, which we don't want

Reply:
I think you'd want to ensure persistence is set to true in this case:
```
const watcher = chokidar.watch('file', {
  persistent: true
});
```

Reply:
Yes we set which folder the watcher should be watched in constructor on `starWatchers`, but we put which logic should apply here, If someone calls 2 time this method just set the same logic to the same event.

`persistent` by default is `true`

Reply:
I changed the code, I move everything related to watchers setup to one private function.